src/parse-user-script.js

```diff
@@ -77,6 +118,9 @@ window.parseUserScript = function(content, url, failIfMissing) {
     case 'noframes':
       details.noFrames = true;
       break;
+    case 'downloadURL':
```

Comment:
It was a very intentional decision to not handle downloadURL, updateURL, etc. in GM4.  I won't merge code that tries to add them again.

This causes a situation that breaks the principle of least astonishment.  I might choose to save a script to my own disk/server, perhaps to isolate it from the original author.  But then by accident through one little setting it downloads/updates to some other remote resource that someone else controls?  Do not want.

Reply:
What do you propose then? We can't get the resources from disk without the user explicitly asking to open a file. As far as I'm aware we can't even open a 'please select next file' picker. Even if we could, asking the user to select every resource is pretty bad UX. Granted, the resources aren't downloaded due to current limitations of `installFromSource` but that should eventually be fixed.

So, should the script be imported without the external resources? (Yes, that's what happens now, but at least with the current PR the URLs are available and can be used to fetch resources).

Everything could be `.zip` but that could be pretty extreme for just a single script.

Alternatively, I could make it so that the original request takes precedence over the `@downloadURL`, that way if you're hosting it somewhere then resources will be fetched from your host, but if you're installing it by just importing the script it can still fetch remote resources using `@downloadURL`.

Or do we just not support relative paths? Which will probably break a lot of compatibility, no?

Reply:
I should also say that some of the above also applies to the full DB import / export. Of course if you're importing an archive created by GM then you have access to everything as it was. However, if you import an archive exported from TM or VM then the only 'bits' that GM understands is the raw script file itself and the value-store (if importing from TM). The previously set, internal, `downloadUrl` will be unavailable (non-existant) and we're stuck with not being able to fetch resources.

Reply:
Not all scripts specify `@downloadURL`, so users would be left with sometimes working sometimes not, even with this.

It would make plenty sense to me to allow: 1) `.user.js` import, if no deps (or all fully-specified and currently downloadable remote refs) and also 2) `.zip` import where there's exactly one `.user.js` and relative-only references to deps otherwise in the zip.

There's also some way to select a directory with a file browser?  I don't know what access you get to files in that directory though.